module/documents/activity/utility.mjs

```diff
@@ -21,7 +21,129 @@ export default class UtilityActivity extends ActivityMixin(UtilityActivityData)
-      sheetClass: UtilitySheet
+      sheetClass: UtilitySheet,
+      usage: {
+        actions: {
+          rollFormula: UtilityActivity.#rollFormula
+        }
+      }
     }, { inplace: false })
   );
+
+  /* -------------------------------------------- */
+  /*  Activation                                  */
+  /* -------------------------------------------- */
+
+  /** @override */
+  _usageChatButtons() {
+    if ( !this.roll.formula ) return null;
+    return [{
+      label: this.roll.label || game.i18n.localize("DND5E.Roll"),
+      icon: '<i class="fa-solid fa-dice" inert></i>',
+      dataset: {
+        action: "rollFormula"
+      }
+    }];
+  }
+
+  /* -------------------------------------------- */
+  /*  Rolling                                     */
+  /* -------------------------------------------- */
+
+  /**
+   * Roll the formula attached to this utility.
+   * @param {BasicRollProcessConfiguration} [config]   Configuration information for the roll.
+   * @param {BasicRollDialogConfiguration} [dialog]    Configuration for the roll dialog.
+   * @param {BasicRollMessageConfiguration} [message]  Configuration for the roll message.
+   * @returns {Promise<BasicRoll[]|void>}              The created Roll instances.
+   */
+  async rollFormula(config={}, dialog={}, message={}) {
+    if ( !this.roll.formula ) {
+      console.warn(`No formula defined for the activity ${this.name} on ${this.item.name} (${this.uuid}).`);
+      return;
+    }
+
+    const rollConfig = foundry.utils.deepClone(config);
+    rollConfig.rolls = [{ parts: [this.roll.formula], data: this.getRollData() }].concat(config.rolls ?? []);
+
+    const dialogConfig = foundry.utils.mergeObject({
+      configure: true
+    }, dialog);
+
+    const messageConfig = foundry.utils.mergeObject({
+      create: true,
+      data: {
+        flavor: `${this.item.name} - ${this.roll.label || game.i18n.localize("DND5E.OtherFormula")}`,
+        flags: {
+          dnd5e: {
+            messageType: "roll",
+            activity: { type: this.type, id: this.id, uuid: this.uuid },
+            item: { type: this.item.type, id: this.item.id, uuid: this.item.uuid }
+          }
+        }
+      }
+    });
+
+    /**
+     * A hook event that fires before a formula is rolled for an Utility activity.
+     * @function dnd5e.preRollFormulaV2
+     * @memberof hookEvents
+     * @param {UtilityActivity} activity               The activity performing the roll.
+     * @param {BasicRollProcessConfiguration} config   Configuration information for the roll.
+     * @param {BasicRollDialogConfiguration} dialog    Configuration for the roll dialog.
+     * @param {BasicRollMessageConfiguration} message  Configuration for the roll message.
+     * @returns {boolean}                   Explicitly return `false` to prevent the roll from being performed.
+     */
+    if ( Hooks.call("dnd5e.preRollFormulaV2", this, rollConfig, dialogConfig, messageConfig) === false ) return;
```

Comment:
Are the hook names placeholders that will later be renamed without the V2 once deprecation is over, or is this permanent?

Reply:
Probably permanent, since removing the V2 from the name would require another long deprecation period, but I suppose after the old format is dropped we could just call both.